frontend/src/views/Lobby.tsx

```diff
@@ -65,36 +66,38 @@ function LobbyPage() {
 
   // Grab the nickname variable and add the user to the lobby.
   useEffect(() => {
-    // Grab the nickname variable; otherwise, set an error.
-    if (location && location.state && location.state.nickname) {
-      setNickname(location.state.nickname);
+    // Grab the user and room information; otherwise, redirect to the join page
+    if (location && location.state && location.state.user && location.state.room) {
+      setUser(location.state.user);
+      setRoom(location.state.room);
     } else {
-      setError('No nickname was provided for the user in the lobby.');
+      setShouldRedirect(true);
```

Comment:
I originally tried that but then realized doing so broke the back button functionality - basically, if you are pushed to the join page, then swipe back, you end up at the lobby page again, which immediately redirects you back to the join page. Idk how big of an issue that is for us though, so I can def change it back if cons outweigh the pros.

Reply:
Ok, that makes sense! I tried replacing this with `window.location.href = '/'`, but the issue you described occurred. I think your solution is good -- my only nit would be to put a comment _briefly_ noting why you are using this instead of the `history` hook.